src/App.tsx

```diff
@@ -1,12 +1,85 @@
 import React from 'react';
 import './App.scss';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { GoodsList } from './components/GoodsList/GoodsList';
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+type State = {
+  goods: Good[];
+  isLoading: boolean;
+  errorMessage: string;
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    goods: [],
+    isLoading: false,
+    errorMessage: '',
+  };
+
+  loadData = async (getData: () => Promise<Good[]>) => {
+    this.setState({
+      isLoading: true,
+    });
```

Comment:
but you need to reset your errorMessage value here, cause otherwise first error would be rendered forever